kimhaejung/app.js

```diff
@@ -0,0 +1,32 @@
+const express = require("express");
+const cors = require("cors");
+const dotenv = require("dotenv");
+const { DataSource } = require("typeorm");
+const app = express();
+const logger = require("morgan");
+
+dotenv.config();
+
+app.use(cors());
+app.use(logger("combined"));
+
+app.get("/ping", function (req, res, next) {
+  res.json({ message: "pong" });
+});
+
+const myDataSource = new DataSource({
+  type: process.env.DB_CONNECTION,
+  host: process.env.DB_HOST,
+  port: process.env.DB_PORT,
+  username: process.env.DB_USERNAME,
+  password: process.env.DB_PASSWORD,
+  database: process.env.DB_DATABASE,
+});
+
+myDataSource.initialize().then(() => {
+  console.log("Data Source has been initialized!");
+});
```

Comment:
`appDataSource.initialize()` 함수가 실행되고 나면, then()과 catch()를 사용해서 에러를 핸들링해줄 수 있습니다. 이렇게 하면, initialize() 함수가 실행되어서 database와 실제로 connection을 생성할 때, 성공했는지 실패했는지 여부를 확인할 수 있습니다.

[공식문서](https://orkhan.gitbook.io/typeorm/docs/data-source)를 참조해주세요!
```suggestion
appDataSource.initialize()
  .then(() => {
    console.log("Data Source has been initialized!");
  })
  .catch((err) => {
    console.error("Error during Data Source initialization", err)
  });
```